ELK stuff

lbjay opened the pull request:
For review. Don't merge.

lbjay committed:
Analytics node setup. Includes:

* add sqs access to instance profile policy doc
* docs for new analytics layer

lbjay commented on README.elk.md:
```diff
@@ -0,0 +1,143 @@
+        "volume_configurations": [{
+            "mount_point": "/vol/data",
+            "number_of_disks": 1,
+            "size": "50",
```
this should maybe be "300" (what you'd want in production)

djcp replied:
That probably makes sense, as we'd most likely want to use this only for production.

lbjay commented on README.elk.md:
```diff
@@ -0,0 +1,143 @@
+
+* An "elasticsearch" layer with one (private) instance running elasticsearch on 
+  port 9200.
+* A "logstash-kibana" lyaer with one (public) instance running:
```
"layer"

djcp commented on README.elk.md:
```diff
@@ -0,0 +1,143 @@
+        }],
+        "instances": {
+            "number_of_instances": 1,
+            "instance_type": "t2.medium",
```
this this big enough? t2.mediums don't get a much RAM and only burstable CPU.

lbjay replied:
oh right. i meant to change that. my earlier testing was on a m4.large which seemed to be sufficient relative to the cost. i'll also try one of the memory enhanced options when i figure out the benchmarking approach.

djcp commented on README.elk.md:
```diff
@@ -0,0 +1,143 @@
+        "instances": {
+            "number_of_instances": 1,
+            "instance_type": "t2.medium",
+            "root_device_type": "instance-store"
```
huh. . . how has this been? Does the instance spin up quickly?  Does this instance have no state worth saving on it at all?

djcp replied:
Also, same question about instance size applies.

lbjay replied:
Nope, no state unless we consider logs under /var/logs worth saving. We don't have those on a volume for other layers, so I assume no. It does seem like it spins up more quickly than the elasticsearch instance, but I haven't been paying that close attention.

lbjay replied:
I guess it should be 'm4.large' too since we want to avoid burstable. Wish there was a lighter option as 8GB seems like a bit much.

djcp replied:
Well, because we're defaultly using ebs-backed instances, logs under /var/log _are_ persistent and stored on EBS volumes. I am definitely willing to experiment, though.

djcp commented on README.analytics.md:
```diff
@@ -0,0 +1,136 @@
+* execute `./bin/rake admin:cluster:init` and `./bin/rake stack:instances:start`
+  to initialize and start the new layer/instance.
+
+## Cluster delete clean-up
```
I'd probably reorder this to be a bit more chronological - or maybe reorganize to make it clear what you have to do in what order to get started.

djcp commented:
Besides my series / type stupidity (and the bug you already fixed around the cluster name), this seems to work well. I'll give another flip through the chef side.

lbjay commented:
OK, I did some rearranging of the readme. Moved the 'Cluster delete...' thing to the end and added "See the Kibana section..." at the end of the setup steps.
